integrations/gcp/gcp_core/search/helpers/ratelimiter/base.py

```diff
@@ -0,0 +1,92 @@
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional, TYPE_CHECKING
+from aiolimiter import AsyncLimiter
+from google.cloud.cloudquotas_v1 import CloudQuotasAsyncClient, GetQuotaInfoRequest
+from loguru import logger
+from enum import Enum
+from port_ocean.context.ocean import ocean
+from gcp_core.search.helpers.ratelimiter.utils import cache_coroutine_result
+
+_DEFAULT_RATE_LIMIT_TIME_PERIOD: float = 60.0
+_DEFAULT_RATE_LIMIT_QUOTA: int = int(
+    ocean.integration_config["search_all_resources_per_minute_quota"]
+)
+
+if TYPE_CHECKING:
+    from google.cloud.cloudquotas_v1 import DimensionsInfo
+
+
+class ContainerType(Enum):
+    PROJECT = "projects"
+    FOLDER = "folders"
+    ORGANIZATION = "organizations"
+
+
+class GCPResourceQuota(ABC):
+    quota_id: str | None = None
+    service: str | None = None
+    container_type: ContainerType = ContainerType.PROJECT
+    _default_quota: int = _DEFAULT_RATE_LIMIT_QUOTA
+
+    async def _request_quota_info(self, name: str) -> List["DimensionsInfo"]:
+        async with AsyncLimiter(
+            max_rate=_DEFAULT_RATE_LIMIT_QUOTA,
+            time_period=_DEFAULT_RATE_LIMIT_TIME_PERIOD,
+        ):
+            async with CloudQuotasAsyncClient() as quotas_client:
+                request = GetQuotaInfoRequest(name=name)
+                response = await quotas_client.get_quota_info(request=request)
+                return list(response.dimensions_infos)
+
+    async def _get_quota(self, container_id: str) -> int:
+        name = self.quota_name(container_id)
+
+        try:
+            quota_infos = await self._request_quota_info(name)
+            quota_info = quota_infos[0] if quota_infos else None
+
+            if not quota_info or quota_info.get("applicable_locations") != ["global"]:
```

Comment:
Doesn't `not quota_info` mean he doesn't have any quota for this container?

Reply:
yes, but specifically, for the asset api's quota,